google/cloud/internal/oauth2_google_credentials_test.cc

```diff
@@ -252,16 +273,25 @@ TEST_F(GoogleCredentialsTest, MissingCredentialsViaGcloudFilePath) {
   // The method to create default credentials should see that no file exists at
   // this path, then continue trying to load the other credential types,
   // eventually finding no valid credentials and hitting a runtime error.
-  ScopedEnvironment gcloud_path_override_env_var(GoogleGcloudAdcFileEnvVar(),
-                                                 filename);
-  ScopedEnvironment gcloud_metadata_host_override_env_var(
+  auto const adc_env = ScopedEnvironment(GoogleGcloudAdcFileEnvVar(), filename);
+  auto const gce_host_env = ScopedEnvironment(
       internal::GceMetadataHostnameEnvVar(), "invalid.google.internal");
 
-  auto creds = GoogleDefaultCredentials();
+  MockHttpClientFactory client_factory;
+  EXPECT_CALL(client_factory, Call).WillRepeatedly([]() {
```

Comment:
should we set any expectations on the `Options const& options` passed to the client factory?

Reply:
Done